.azure/templates/jobs/build_java.yaml

```diff
@@ -39,11 +36,34 @@ jobs:
           BRANCH: $(Build.SourceBranch)
           TESTCONTAINERS_RYUK_DISABLED: "TRUE"
           TESTCONTAINERS_CHECKS_DISABLE: "TRUE"
-          MVN_ARGS: "-e -V -B"
       - bash: "mvn ${MVN_ARGS} spotbugs:check"
         displayName: "Spotbugs"
-        env:
-          MVN_ARGS: "-e -V -B"
+      - bash: "mvn ${MVN_ARGS} clean install -f examples/docker ${MVN_EXTRA_ARGS}"
+        displayName: "Test Examples"
+      - bash: |
+          cd examples/docker
+          ./spring/test-spring.sh
+        displayName: "Test Spring Example"
+        condition: and(succeeded(), eq(variables['jdk_version'], '17'))
```

Comment:
Do we need this? What is the purpose of it?

Reply:
The Spring Example test requires Java 17. The `succeeded()` function means that it will only execute if the previous step was successful (the compilation).